src/pages/Admin/Sidebar/SidebarOpener/useSidebarOpener.ts

```diff
@@ -0,0 +1,45 @@
+import { matchPath, useLocation } from "react-router-dom";
+import { Link, LinkGroup } from "../types";
+import { useModalContext } from "contexts/ModalContext";
+import useHandleScreenResize, { SCREEN_MD } from "hooks/useHandleScreenResize";
+import { appRoutes } from "constants/routes";
+import ToggleableSidebar from "./ToggleableSidebar";
+
+const ADMIN_ROUTE = `${appRoutes.admin}/:id/`;
+
+const DEFAULT_LINK: Link = {
+  title: "Open Menu",
+  icon: { type: "Menu", size: 24 },
+  to: "",
+};
+
+export default function useSidebarOpener(linkGroups: LinkGroup[]) {
+  const currPath = useLocation().pathname;
+
+  // Explanation for the `reduce()` part:
+  // Since `matchPath` returns all paths that match the pattern and that
+  // includes all the parent paths, but we want to return only the "first parent" link
+  // (the one with more path segments), we can find that "first parent" by simply checking the
+  // `to` field length -> higher the length, more "recent" the parrent
+  const activeLink =
+    linkGroups
+      .flatMap((g) => g.links)
+      .filter((link) => !!matchPath(`${ADMIN_ROUTE}${link.to}/*`, currPath))
```

Comment:
could go straight to `reduce` just add another data to `prev` e.g `[Link, actualPath]` and use whether to update or not
```suggestion
      .reduce((link) => !!matchPath(`${ADMIN_ROUTE}${link.to}/*`, currPath))
```